pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,382 @@
+	}
+	return true, nil
+}
+
+func parseFlow() (*v1alpha1.Packet, error) {
+	cleanFlow := strings.ReplaceAll(option.flow, " ", "")
+	pkt := new(v1alpha1.Packet)
+
+	for _, v := range strings.Split(cleanFlow, ",") {
+		n, ok := protocols[v]
+		if ok {
+			(*pkt).IPHeader.Protocol = n
+			break
+		}
+	}
+
+	r, ok, err := getFieldPortValue(cleanFlow, "tcp_src")
+	if err != nil {
+		return nil, fmt.Errorf("error when get tcp_src value: %w", err)
+	}
+	if ok {
+		if (*pkt).TransportHeader.TCP == nil {
+			(*pkt).TransportHeader.TCP = new(v1alpha1.TCPHeader)
+		}
+		(*pkt).TransportHeader.TCP.SrcPort = int32(r)
+	}
+	r, ok, err = getFieldPortValue(cleanFlow, "tcp_dst")
+	if err != nil {
+		return nil, fmt.Errorf("error when get tcp_dst value: %w", err)
+	}
+	if ok {
+		if (*pkt).TransportHeader.TCP == nil {
+			(*pkt).TransportHeader.TCP = new(v1alpha1.TCPHeader)
+		}
+		(*pkt).TransportHeader.TCP.DstPort = int32(r)
+	}
+	r, ok, err = getFieldPortValue(cleanFlow, "udp_src")
+	if err != nil {
+		return nil, fmt.Errorf("error when get udp_src value: %w", err)
+	}
+	if ok {
+		if (*pkt).TransportHeader.UDP == nil {
+			(*pkt).TransportHeader.UDP = new(v1alpha1.UDPHeader)
+		}
+		(*pkt).TransportHeader.UDP.SrcPort = int32(r)
+	}
+	r, ok, err = getFieldPortValue(cleanFlow, "udp_dst")
+	if err != nil {
+		return nil, fmt.Errorf("error when get udp_dst value: %w", err)
+	}
+	if ok {
+		if (*pkt).TransportHeader.UDP == nil {
+			(*pkt).TransportHeader.UDP = new(v1alpha1.UDPHeader)
+		}
+		(*pkt).TransportHeader.UDP.DstPort = int32(r)
+	}
+
+	return pkt, nil
+}
+
+func getFieldPortValue(cleanFlow string, f string) (int, bool, error) {
+	for _, v := range strings.Split(cleanFlow, ",") {
+		m, err := regexp.MatchString(fmt.Sprintf("%s=[0-9]+", f), v)
+		if err != nil {
+			return 0, false, err
+		}
+		if m {
+			r, err := strconv.Atoi(v[len(f)+1:])
+			if err != nil {
+				return 0, false, err
+			}
+			return r, true, nil
+		}
+	}
+	return 0, false, nil
+}
+
+func output(tf *v1alpha1.Traceflow) error {
+	r := Response{
+		Name:        tf.Name,
+		Phase:       tf.Status.Phase,
+		Source:      fmt.Sprintf("%s/%s", tf.Spec.Source.Namespace, tf.Spec.Source.Pod),
+		Destination: tf.Spec.Destination.IP,
+		NodeResults: tf.Status.Results,
+	}
+	if len(tf.Spec.Destination.IP) == 0 {
+		if len(tf.Spec.Destination.Service) != 0 {
+			r.Destination = fmt.Sprintf("%s/%s", tf.Spec.Destination.Namespace, tf.Spec.Destination.Service)
+		} else {
+			r.Destination = fmt.Sprintf("%s/%s", tf.Spec.Destination.Namespace, tf.Spec.Destination.Pod)
+		}
+	}
+	if option.outputType == "json" {
+		if err := jsonOutput(&r); err != nil {
+			return fmt.Errorf("error when converting output to json: %w", err)
+		}
+	} else if option.outputType == "yaml" {
+		if err := yamlOutput(&r); err != nil {
+			return fmt.Errorf("error when converting output to yaml: %w", err)
+		}
+	} else {
+		return fmt.Errorf("output types are yaml and json")
```

Comment:
The error is not clear

Reply:
Turned it into "output types should be yaml or json".